language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/formatting/FormattingNodeTree.java

```diff
@@ -2969,6 +2969,56 @@ public void formatRecordTypeNode(JsonObject node) {
 
             if (node.has(FormattingConstants.WS)) {
                 JsonArray ws = node.getAsJsonArray(FormattingConstants.WS);
+                boolean isSealed = node.has("sealed") && node.get("sealed").getAsBoolean();
+                String indentation = this.getIndentation(formatConfig, false);
+
+                this.preserveHeight(ws, indentation);
+
+                boolean firstPipeVisited = false;
+                for (JsonElement wsItem : ws) {
+                    JsonObject currentWS = wsItem.getAsJsonObject();
+                    if (this.noHeightAvailable(currentWS.get(FormattingConstants.WS).getAsString())) {
+                        String text = currentWS.get(FormattingConstants.TEXT).getAsString();
+                        if (text.equals("record")) {
+                            if (formatConfig.get(FormattingConstants.NEW_LINE_COUNT).getAsInt() > 0) {
+                                currentWS.addProperty(FormattingConstants.WS,
+                                        this.getNewLines(formatConfig.get(FormattingConstants.NEW_LINE_COUNT)
+                                                .getAsInt()) + indentation);
+                            } else {
+                                currentWS.addProperty(FormattingConstants.WS, FormattingConstants.SINGLE_SPACE);
+                            }
+                        } else if (text.equals("{")) {
```

Comment:
Better to move these to constants.

Reply:
@NipunaMarcus FYI

Reply:
@hasithaa working on it  as i'm doing a refactoring ....